packages/examples/phases/src/utils.tsx

```diff
@@ -0,0 +1,25 @@
+import React, {FC, useEffect} from 'react';
+
+export function deferredLoadComponent<T>(
+  Resource: DeferredImport<T extends {default: any} ? T : never>,
+): FC {
+  let loaded = false;
+  let cleanUp: undefined | (() => void);
+  return function WrappedComponent(props) {
+    useEffect(() => {
+      return () => {
+        cleanUp?.();
```

Comment:
Important otherwise we get memory leaks, no keen on this as it looks like a dangerous pattern. While people probably don't want to be using the imports directly often, I'd be happy if someone can suggest a better way of cleaning up automatically